examples/mocks/test/automocking.spec.ts

```diff
@@ -40,13 +41,25 @@ test('all mocked are valid', async() => {
   expect(example.symbol).toEqual(Symbol.for('a.b.c'))
 })
 
-test('automock doesn\'t throw when unmocked', () => {
-  // logger uses symbols on its prototype
-  // we are checking here that after unmocking
-  // these symbols are accessible
+test('automock properly restores mock', async() => {
+  expect(log.warn()).toBeUndefined()
+  expect(moduleWithSymbol.warn()).toBeUndefined()
+  expect(moduleWithSymbol[methodSymbol]()).toBeUndefined()
+
+  vi.restoreAllMocks()
+
+  // `spies` (on linux) is an empty Set()
```

Comment:
Hm, it seems `ensureSpy` is not working properly on Linux. Can you go deep into that?

This function should import the same module that Vitest uses, so `spies` will refer to the same Set. Maybe the path is incorrect on Linux, or it has double slashes or something?

I would start with logging the whole `moduleCache` to see if it has duplicate spy module.

Reply:
If not, I will try to find a time and look into it myself.

Reply:
So, I found some time, and this changed fixed it:

https://github.com/vitest-dev/vitest/blob/8b3d4128f01d6f046637679ab9b073add052a5fe/packages/vitest/src/runtime/mocker.ts#L246

```ts
    VitestMocker.spyModule = await this.request(
      `/@fs/${slash(resolve(distDir, 'spy.js'))}`,
    ) as typeof import('../integrations/spy')
    ```

Reply:
thanks, that works